internal/component/database_observability/mysql/collector/explain_plans.go

```diff
@@ -46,28 +46,18 @@ const selectDigestsForExplainPlan = `
 
 const selectExplainPlanPrefix = `EXPLAIN FORMAT=JSON `
 
-func newExplainPlansOutput(logger log.Logger, dbVersion string, digest string, explainJson []byte, generatedAt string) (*database_observability.ExplainPlanOutput, error) {
-	output := &database_observability.ExplainPlanOutput{
-		Metadata: database_observability.ExplainPlanMetadataInfo{
-			DatabaseEngine:  "MySQL",
-			DatabaseVersion: dbVersion,
-			QueryIdentifier: digest,
-			GeneratedAt:     generatedAt,
-		},
-	}
-
+func newExplainPlansOutput(logger log.Logger, explainJson []byte) (*database_observability.ExplainPlanNode, error) {
 	qblock, _, _, err := jsonparser.Get(explainJson, "query_block")
 	if err != nil {
-		return output, fmt.Errorf("failed to get query block: %w", err)
+		return nil, fmt.Errorf("failed to get query block: %w", err)
 	}
 
 	planNode, err := parseTopLevelPlanNode(logger, qblock)
-	output.Plan = planNode
 	if err != nil {
-		return output, err
+		return nil, err
 	}
 
-	return output, nil
+	return &planNode, nil
```

Comment:
Now that this method has been cut down a lot, perhaps we could just `return parseTopLevelPlanNode(logger, qblock)`?